src/lib/annotations/doc/DocDrawingDialog.js

```diff
@@ -1,41 +1,254 @@
+            this.commitButtonEl.removeEventListener('click', this.postDrawing);
+            this.commitButtonEl.removeEventListener('touchend', this.postDrawing);
+        }
+
+        if (this.deleteButtonEl) {
+            this.deleteButtonEl.removeEventListener('click', this.deleteAnnotation);
+            this.deleteButtonEl.removeEventListener('touchend', this.deleteAnnotation);
+        }
+    }
+
+    /**
      * Sets up the drawing dialog element.
      *
+     * @protected
      * @param {Annotation[]} annotations - Annotations to show in the dialog
      * @param {HTMLElement} threadEl - Annotation icon element
      * @return {void}
+     */
+    setup(annotations) {
+        // Create outermost element container
+        this.element = document.createElement('div');
+        this.element.addEventListener('click', annotatorUtil.prevDefAndStopProp);
+        this.element.classList.add(constants.CLASS_ANNOTATION_DIALOG);
+
+        // Create the dialog element consisting of a label, save, and delete button
+        this.drawingDialogEl = this.generateDialogEl(annotations);
+
+        // Set the newly created buttons from the dialog element
+        this.commitButtonEl = this.drawingDialogEl.querySelector(`.${constants.CLASS_ADD_DRAWING_BTN}`);
+        this.deleteButtonEl = this.drawingDialogEl.querySelector(`.${constants.CLASS_DELETE_DRAWING_BTN}`);
+
+        this.bindDOMListeners();
+
+        if (annotations.length > 0) {
+            this.assignDrawingLabel(annotations[0]);
+        }
+
+        this.element.appendChild(this.drawingDialogEl);
+    }
+
+    /**
+     * Position the drawing dialog with an x,y browser coordinate
+     *
+     * @protected
+     * @param {number} x - The x position to position the dialog with
+     * @param {number} y - The y position to position the dialog with
+     * @return {void}
+     */
+    position(x, y) {
+        if (!this.pageEl) {
+            this.pageEl = this.annotatedElement.querySelector(`[data-page-number="${this.location.page}"]`);
+        }
+
+        // Reinsert when the dialog is removed from the page
+        if (!this.pageEl.contains(this.element)) {
+            this.pageEl.appendChild(this.element);
+        }
+
+        // Show dialog so we can get width
+        const clientRect = this.element.getBoundingClientRect();
+        this.element.style.left = `${x - clientRect.width}px`;
+        this.element.style.top = `${y}px`;
+    }
+
+    /**
+     * Hide the dialog in the browser
+     *
      * @protected
+     * @return {void}
      */
-    /* eslint-disable no-unused-vars */
-    setup(annotations, threadEl) {}
-    /* eslint-enable no-unused-vars */
+    hide() {
+        annotatorUtil.hideElement(this.element);
+        this.visible = false;
+    }
 
     /**
      * Display the dialog in the browser
      *
-         * @protected
+     * @protected
+     * @return {void}
+     */
+    show() {
+        annotatorUtil.showElement(this.element);
```

Comment:
Is the mobile dialog going to be the same as the desktop one? See the show/hide methods in AnnotationDialog.js about mobile dialog specific logic

Reply:
Yep, mobile and desktop dialog will be the same.